libvast/src/arrow_table_slice.cpp

```diff
@@ -1141,4 +1146,171 @@ template class arrow_table_slice<fbs::table_slice::arrow::v0>;
 template class arrow_table_slice<fbs::table_slice::arrow::v1>;
 template class arrow_table_slice<fbs::table_slice::arrow::experimental>;
 
+// -- utility functions --------------------------------------------------------
+
+namespace {
+
+std::shared_ptr<arrow::Array>
+convert_subnet_array(const arrow::FixedSizeBinaryArray& arr) {
+  auto* pool = arrow::default_memory_pool();
+  auto subnet_builder = subnet_type::make_arrow_builder(pool);
+  auto& address_builder = subnet_builder->address_builder();
+  auto& length_builder = subnet_builder->length_builder();
+  if (!address_builder.Resize(arr.length()).ok()
+      || !length_builder.Resize(arr.length()).ok()
+      || !subnet_builder->Resize(arr.length()).ok())
+    die("unable to resize builders to target capacity");
+  for (int row = 0; row < arr.length(); ++row) {
+    if (arr.IsNull(row)) {
+      if (!subnet_builder->AppendNull().ok())
+        die("append null value to subnet builder failed");
+    } else {
+      auto span = std::span<const uint8_t, 17>{arr.GetValue(row), 17};
+      if (!(subnet_builder->Append().ok()
+            && address_builder.Append(arr.GetValue(row)).ok()
+            && length_builder.Append(span[16]).ok()))
+        die("rebuilding of subnet struct array failed");
+    }
+  }
+  if (auto result = subnet_builder->Finish(); result.ok())
+    return *result;
+  die("failed to finish Arrow subnet column builder");
+}
+
+// TODO: this is copied from
+// experimental_table_slice_builder/enum_column_builder, can this live somewhere
+// shared?
+std::shared_ptr<arrow::Array> make_dict_array(const enumeration_type& et) {
```

Comment:
This function should not be needed after the other recommended changes.

Reply:
from the resize docs:
```
  /// \brief Ensure that enough memory has been allocated to fit the indicated
  /// number of total elements in the builder, including any that have already
  /// been appended.
```

`Reserve` actually calls `Resize`, so I think that I'm using the right function, just the names are somewhat unconventional :)

Reply:
It's finally gone, thanks for helping with this.